### DISCUSSION TITLE
Prevent xyzRow.ts generation

### DISCUSSION THREAD

**Discussion by marcobisio** [serenity-is/Serenity#6568] [Help] [answered]
Hi everyone,
I was wondering if there is a way to prevent server typings generation for a single xyzRow.cs or, even better, prevent generation for a single property. I tried with ScriptSkip, JsonIgnore and so on, but with no luck.
Any suggestion?

Thanks

**Comment by volkanceylan** [comment, marked as answer]
Types with ScriptSkip are normally not generated, but if they are referenced somewhere like endpoints, response types etc. they will still get generated. Maybe we should still check ScriptSkip for referenced types but in that case, the code generated for classes which references them would result in typescript errors, so would also need to set them to type "any". I'll add this to backlog.

Meanwhile, not sure if it would work, but try setting the property visibility to "internal" or "protected".

**Reply by marcobisio** [reply]
"internal" seems to do the job, thanks!

**Comment by volkanceylan** [comment]
6.2.9 should allow you to not generate code for rows by [ScriptSkip] even if it is referenced, same for row members. It was a quick fix, may result in some other issues not sure.

**Comment by marcobisio** [comment]
Thanks for trying @volkanceylan .
I just put ScriptSkip in a xyzRow.cs as a class attribute, but it doesn't work: the related ts is being still generated.
I also tried to manually deleted the xyzRow.ts, but it is generated back.

**Comment by volkanceylan** [comment]
Have you `dotnet tool update sergen`?, and update Pro.Coder if using source generators? Also restart VS, as source generators are not updated by VS

**Comment by marcobisio** [comment]
Ah, only upgraded the Pro.Coder in the previous try...just upgraded also sergen and now it's working properly, thanks again @volkanceylan !
